calico_kubernetes/calico_kubernetes.py

```diff
@@ -497,217 +540,53 @@ def _api_root_secure(self):
-        """
-        In addition to Calico's default pod_name tag,
-        Add tags generated from Kubernetes Labels and Namespace
-            Ex. labels: {key:value} -> tags+= namespace_key_value
-        Add tag for namespace
-            Ex. namespace: default -> tags+= namespace_default
-
-        :param self.profile_name: The name of the Calico profile.
-        :type self.profile_name: string
-        :param pod: The config dictionary for the pod being created.
-        :type pod: dict
-        :return:
-        """
-        logger.info('Applying tags')
-
-        try:
-            profile = self._datastore_client.get_profile(self.profile_name)
-        except KeyError:
-            logger.error('Could not apply tags. Profile %s could not be found. Exiting', self.profile_name)
+            logger.error(
+                "Could not apply rules. Profile not found: %s, exiting", self.profile_name)
             sys.exit(1)
 
-        # Grab namespace and create a tag if it exists.
-        ns_tag = self._get_namespace_tag(pod)
-
-        if ns_tag:
-            logger.info('Adding tag %s' % ns_tag)
-            profile.tags.add(ns_tag)
-        else:
-            logger.warning('Namespace tag cannot be generated')
-
-        # Create tags from labels
-        labels = self._get_metadata(pod, 'labels')
-        if labels:
-            for k, v in labels.iteritems():
-                tag = self._label_to_tag(k, v)
-                logger.info('Adding tag ' + tag)
-                profile.tags.add(tag)
+        # Determine rule set based on policy
+        if self.profile_name == "REJECT_ALL":
+            default_rule = Rule(action="deny")
+            logger.info("Using deny all rules")
         else:
-            logger.warning('No labels found in pod %s' % pod)
-
-        self._datastore_client.profile_update_tags(profile)
-
-        logger.info('Finished applying tags.')
-
-    def _get_metadata(self, pod, key):
-        """
-        Return Metadata[key] Object given Pod
-        Returns None if no key-value exists
-        """
-        try:
-            val = pod['metadata'][key]
-        except (KeyError, TypeError):
-            logger.warning('No %s found in pod %s', key, pod)
-            return None
-
-        logger.debug("%s of pod %s:\n%s", key, pod, val)
-        return val
-
-    def _escape_chars(self, unescaped_string):
-        """
-        Calico can only handle 3 special chars, '_.-'
-        This function uses regex sub to replace SCs with '_'
-        """
-        # Character to replace symbols
-        swap_char = '_'
+            default_rule = Rule(action="allow")
+            logger.info("Using allow all rules")
 
-        # If swap_char is in string, double it.
-        unescaped_string = re.sub(swap_char, "%s%s" % (swap_char, swap_char), unescaped_string)
+        rules = Rules(id=profile.name,
+                      inbound_rules=[default_rule],
+                      outbound_rules=[default_rule])
```

Comment:
As discussed, outbound_rule should always be "allow".

The CALICO_POLICY flag only controls the inbound rules.